blueocean-config/src/main/java/io/jenkins/blueocean/config/BlueOceanConfigFactory.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.blueocean.config;
+
+import javax.annotation.Nonnull;
+
+import hudson.ExtensionPoint;
+import jenkins.model.Jenkins;
+
+/**
+ * Blue Ocean configuration factory which allows different extensions to override the default behavior.
+ * 
+ */
+public abstract class BlueOceanConfigFactory implements ExtensionPoint {
+    public static final String FEATURE_PROPERTY_PREFIX = "blueocean.features.";
+    public static final String ORGANIZATIONS_ENABLED = "organizations.enabled";
+
+    private static BlueOceanConfig config;
+
+    /**
+     * Extension method to be implemented for each Blue Ocean configuration factory. Any value left <code>null</code> on
+     * the <code>config</code> parameter will be initialized by {@link BlueOceanConfigurationFactoryDefault},
+     * initialized values won't be overriden.
+     * 
+     * @param config to initialize
+     * @return the initialized {@link BlueOceanConfig}
+     */
+    @Nonnull
+    protected abstract BlueOceanConfig configure(@Nonnull BlueOceanConfig config);
```

Comment:
This is redundant. With higher ordinal you can override method such as 'BlueoceanConfig getConfig()'.